### PR TITLE
Copy and Paste

### PR DESCRIPTION
**Pull request by qwertynerd97**
Improves paste features in the editor.
Users can now copy Items in the editor and then paste them back in, without having to go through a secondary text editor.  
Users can copy the content of one MC Choice and then paste it into another
Users can copy and paste within the Assessment section
Upgraded to a minor fix of Slate, so users can now cut within the OboEditor.
Adds the supportsChildren parameter to OboNode registration; OboNodes that support children will reduce the tree structure to leaf-most OboNodes when pasting.  This prevents users from pasting Questions inside themselves and causing errors.

Partially implements a solution to https://github.com/ucfopen/Obojobo/issues/724
Does not include merging paste

### CODE REVIEW

File: packages/obojobo-document-engine/ObojoboDraft/Chunks/Table/schema.js [hunk 1]
```diff
@@ -21,6 +21,11 @@ const schema = {
 				const header = index === 0 && node.data.get('content').header
 				switch (error.code) {
 					case CHILD_TYPE_INVALID: {
+						// Deal with Paste insertion of top-level nodes
```
[the remark this answers is not in the document]
Comment (zachberry) [reply packages/obojobo-document-engine/ObojoboDraft/Chunks/Table/schema.js]: Agree

File: packages/obojobo-document-engine/__tests__/ObojoboDraft/Chunks/Code/editor.test.js [hunk 2]
```diff
@@ -424,6 +420,21 @@ describe('Code editor', () => {
...
 	})
 
+	test('plugins.schema.normalize fixes invalid oboeditor.component in code', () => {
+		const editor = {
+			unwrapNodeByKey: jest.fn()
+		}
+
+		Code.plugins.schema.blocks[CODE_NODE].normalize(editor, {
+			code: CHILD_TYPE_INVALID,
+			node: { nodes: { size: 10 } },
```
Comment (zachberry) [inline packages/obojobo-document-engine/__tests__/ObojoboDraft/Chunks/Code/editor.test.js]: What does the "size 10" bit mean here?

**Comment by qwertynerd97** [comment]
Talked about IRL: 
Right now, pasting Obojobo Editor nodes into other nodes creates a new node every time.  In the future, we would like to be able to paste a node and have it merge into the existing node.  Currently, the way Slate's command system is set up, this is very difficult.  Slate changed their command structure in 0.45 (as well as breaking a couple of other things), so when they release 0.46 we might want to upgrade and take another look at how we are doing our pasting.

**Comment by zachberry** [comment]
User comments who experienced a bug with the current editor. Want to test to see if these changes fix these issues:

```
Running into a bug with the editor
When I select the title of a page and hit CTrl + V trying to paste some text, the editor crashes
this seems to happen only when I select the entire title, and hit Ctrl + V
and often happens on the second attempt
on the first attempt, the text I copied will be pasted in the middle of the existing text, instead of replacing it.
then on the second attempt the whole thing crashed
re-freshing the page seems to be able to restore the editor
and selecting part of the title doesn't cause the bug

Only affects headings of any level (edited) 
```

**Comment by zachberry** [comment]
I tried the above ^ - it works. The behavior has changed so that bug doesn't occur. I think we still want to move towards the behavior mentioned in the last post by @qwertynerd97

File: packages/obonode/obojobo-sections-assessment/schema.js [hunk 3]
```diff
@@ -25,6 +25,7 @@ const schema = {
 				{ match: [{ type: RUBRIC_NODE }] }
 			],
 			normalize: (editor, error) => {
+				console.log('assessment', error)
```
Comment (zachberry) [inline packages/obonode/obojobo-sections-assessment/schema.js]: Left in console
```suggestion
```

**Comment by zachberry** [comment]
Spending some more time trying to break things. I can't figure out a consistent pattern yet, but these are the things I've found:

* [ ] - Sometimes using Ctrl+A to select all then pressing delete doesn't delete all the content. Instead it deletes a single character where the cursor was.
* [ ] - Sometimes you can't use Ctrl+A, instead the selection flashes then disappears. When this happens you need to change pages to restore normal Ctrl+A functionality.
* [ ] - Pasting in complicated content, like big HTML pages or large Obojobo pages, still can fire errors in the editor and the past doesn't go through. So far however these haven't killed the editor, it seems to be able to continue working.

I don't want to block these enhancements further so I'm going to move the above into a new issue, but I wanted to mention it here regardless.

**Pull request merged**